src/shared/image-policy.h

```diff
@@ -0,0 +1,95 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+#pragma once
+
+typedef struct ImagePolicy ImagePolicy;
+
+#include "dissect-image.h"
+#include "errno-list.h"
+
+typedef enum PartitionPolicyFlags {
+        /* Not all policy flags really make sense on all partition types, see comments. But even if they
+         * don't make sense we'll parse them anyway, because maybe one day we'll add them for more partition
+         * types, too. Moreover, we allow configuring a "default" policy for all partition types for which no
+         * explicit policy is specified. It's useful if we can use policy flags in there and apply this
+         * default policy gracefully even to partition types where they don't really make too much sense
+         * on. Example: a default policy of "verity+encrypted" certainly makes sense, but for /home/
+         * partitions this gracefully degrades to "encrypted" (as we do not have a concept of verity for
+         * /home/), and so on. */
+        PARTITION_POLICY_VERITY               = 1 << 0, /* must exist, use with verity                 (only applies to root/usr partitions) */
+        PARTITION_POLICY_SIGNED               = 1 << 1, /* must exist, use with signed verity          (only applies to root/usr partitions) */
+        PARTITION_POLICY_ENCRYPTED            = 1 << 2, /* must exist, use with LUKS encryption        (applies to any data partition, but not to verity/signature partitions */
+        PARTITION_POLICY_UNPROTECTED          = 1 << 3, /* must exist, use without encryption/verity */
+        PARTITION_POLICY_UNUSED               = 1 << 4, /* must exist, don't use */
+        PARTITION_POLICY_ABSENT               = 1 << 5, /* must not exist */
+        PARTITION_POLICY_OPEN                 = PARTITION_POLICY_VERITY|PARTITION_POLICY_SIGNED|PARTITION_POLICY_ENCRYPTED|PARTITION_POLICY_UNPROTECTED|PARTITION_POLICY_UNUSED|PARTITION_POLICY_ABSENT,
+        PARTITION_POLICY_IGNORE               = PARTITION_POLICY_UNUSED|PARTITION_POLICY_ABSENT,
+        _PARTITION_POLICY_USE_MASK            = PARTITION_POLICY_OPEN,
+
+        PARTITION_POLICY_READ_ONLY_OFF        = 1 << 6, /* State of GPT partition flag "read-only" must be on */
+        PARTITION_POLICY_READ_ONLY_ON         = 1 << 7,
+        _PARTITION_POLICY_READ_ONLY_MASK      = PARTITION_POLICY_READ_ONLY_OFF|PARTITION_POLICY_READ_ONLY_ON,
+        PARTITION_POLICY_GROWFS_OFF           = 1 << 8, /* State of GPT partition flag "growfs" must be on */
+        PARTITION_POLICY_GROWFS_ON            = 1 << 9,
+        _PARTITION_POLICY_GROWFS_MASK         = PARTITION_POLICY_GROWFS_OFF|PARTITION_POLICY_GROWFS_ON,
+        _PARTITION_POLICY_PFLAGS_MASK         = _PARTITION_POLICY_READ_ONLY_MASK|_PARTITION_POLICY_GROWFS_MASK,
```

Comment:
the reason i used `*`, `-`, `~` was mostly because I wanted to communicate clearly that these aren't partition flags, but shortcuts for *whole* full-image policies. i.e. they should look substantially different than regular flag specs. hence I opted for special chars instead of english words, because english words we already use for partition flags.

Moreover, if you specify no flags at all (i.e. an empty string policy), then nothing is used ("ignore" policy"). In various tools we already accept "-" in place of an empty string, to make things easier to read. Hence I figured it would make sense to follow this same logic here: the empty string policy is the same as "-". (the man page only mentions the latter for simplicity)

From there "*" and "~" where logical next steps for the permissive and the prohibitive policies.

That said, I'd be OK with renaming these to `@ignoreall`, `@prohibitall` and `@permitall` or so, if you insist. key would be to use a syntax that is sufficiently different from the flags syntax we otherwise use. (but quite frankly, i still like the "-", "~", "*" thing)

Reply:
And "*" looks indeed like a wildcard match, and that's exactly what it is supposed to be: it means "use everything there is". So this is not coincidental, this is on purpose